connector/src/main/java/org/geysermc/connector/network/translators/item/ItemTranslator.java

```diff
@@ -125,12 +132,37 @@ public ItemData translateToBedrock(GeyserSession session, ItemStack stack) {
             }
         }
 
+        ItemData itemData;
         ItemStackTranslator itemStackTranslator = itemTranslators.get(bedrockItem.getJavaId());
         if (itemStackTranslator != null) {
-            return itemStackTranslator.translateToBedrock(itemStack, bedrockItem);
+            itemData = itemStackTranslator.translateToBedrock(itemStack, bedrockItem);
         } else {
-            return DEFAULT_TRANSLATOR.translateToBedrock(itemStack, bedrockItem);
+            itemData = DEFAULT_TRANSLATOR.translateToBedrock(itemStack, bedrockItem);
         }
+
+        try {
+            // Get the display name of the item
+            String name = itemData.getTag().getCompound("display").getString("Name");
+
+            // Check if its a message to translate
+            if (MessageUtils.isMessage(name)) {
+                // Get the translated name
+                name = MessageUtils.getTranslatedBedrockMessage(Message.fromString(name), session.getClientData().getLanguageCode());
+
+                // Build the new display tag
+                CompoundTagBuilder displayBuilder = itemData.getTag().getCompound("display").toBuilder();
+                displayBuilder.stringTag("Name", name);
+
+                // Build the new root tag
+                CompoundTagBuilder builder = itemData.getTag().toBuilder();
+                builder.tag(displayBuilder.build("display"));
+
+                // Create a new item with the original data + updated name
+                itemData = ItemData.of(itemData.getId(), itemData.getDamage(), itemData.getCount(), builder.buildRootTag());
+            }
+        } catch (Exception e) { }
```

Comment:
Why an all-purpose "catch exception"?

Reply:
It was left in, from some other code. But I thought it may need it for the name getting if the nbt is null `itemData.getTag()` will return an NPE.

Reply:
It would likely be cleaner and more self-explanatory for test for an NPE.

Reply:
Or, at the least, add a debug ;ine here.